src/App.jsx

```diff
@@ -14,45 +19,97 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const Sort_Letter = 'FirstLetter';
+const Sort_Length = 'Length';
+
+function getPreparedGoods(goods, { sortField, isReversed }) {
+  let prepearedGoods = [...goods];
+
+  if (sortField) {
+    prepearedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case Sort_Letter:
+          return good1.localeCompare(good2);
+        case Sort_Length:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    })
+  }
+
+  if (isReversed) {
+    prepearedGoods.reverse();
+  }
+  
+  return prepearedGoods;
+}
+
+export const App = () => {
+
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, { sortField, isReversed });
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+
+  const reset = () => {
+    setSortField('');
+    setIsReversed(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          onClick={() => {
+            setSortField(Sort_Letter);
+          }}
+          className={cn("button is-info", { 'is-light': sortField !== Sort_Letter })}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn("button is-success", sortField !== Sort_Length ? "is-light" : "")}
```

Comment:
<img width="532" alt="image" src="https://github.com/mate-academy/react_list-of-goods-js/assets/70220667/3e1f7a52-92e2-418b-abdf-fbdfc4fee846">